src/GitVersion.Core/Core/GitPreparer.cs

```diff
@@ -183,6 +183,11 @@ private void NormalizeGitDirectory(bool noFetch, string? currentBranchName, bool
 
         EnsureHeadIsAttachedToBranch(currentBranchName, authentication);
         EnsureRepositoryHeadDuringNormalisation(nameof(EnsureHeadIsAttachedToBranch), expectedSha);
+
+        if (this.repository.IsShallow)
+        {
+            throw new WarningException("Repository is a shallow clone. Git repositories must be contain the full history.");
```

Comment:
The [Requirements page](https://gitversion.net/docs/reference/requirements) actually already says:
> The repository needs to be an [unshallow](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---unshallow) clone. This means that the fetch-depth in GitHub Actions needs to be set to 0, for instance. Check with your [build server](https://gitversion.net/docs/reference/build-servers) to see how it can be configured appropriately.

Maybe it's good enough to link there?

However, the new changes in Azure DevOps from #3188 which triggered this pull request aren't described in [the build servers section about Azure DevOps](https://gitversion.net/docs/reference/build-servers/azure-devops), so maybe we can also add an alert box there [like the one for BitBucket](https://gitversion.net/docs/reference/build-servers/bitbucket-pipelines) which include instructions or, maybe easier yet, link to [the DevOps documentation on the subject](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#shallow-fetch)?

Reply:
I have added a link to the error message and an alert box for Azure DevOps in the documentation. Let me know what you think.